File: Dev/C++/Mascaret/src/HAVE/VirtualHuman.cpp
#include "HAVE/VirtualHuman.h"
#include "VEHA/Entity/ShapeSpecification.h"
#include "HAVE/Body.h"


namespace HAVE {

VirtualHuman::VirtualHuman(shared_ptr<AgentPlateform> ap, const string & name, shared_ptr<AgentClass> agentclass) : EmbodiedAgent(ap, name, agentclass)
{
}

VirtualHuman::~VirtualHuman()
{}

bool VirtualHuman::playAnimation(const string& animationName, int sens, double animationSpeed, bool cycle)
{
	return getBody()->playAnimation(animationName,sens,animationSpeed, cycle);
}
bool VirtualHuman::stopAnimation(const string& animationName)
{
	return getBody()->stopAnimation(animationName);
}
bool VirtualHuman::stopAllAnimations()
{
	return getBody()->stopAllAnimations();
}
bool VirtualHuman::playSound(const string& soundName, double pitch, bool cycle)
{
	return getBody()->playSound(soundName,pitch, cycle);
}
bool VirtualHuman::stopSound(const string& soundName)
{
	return getBody()->stopSound(soundName);
}
}

File: Dev/CS/Mascaret/Mascaret/HAVE/VirtualHuman.cs
using System;
using System.Collections.Generic;
//using TextToSpeech;

namespace Mascaret
{
    public class VirtualHuman : EmbodiedAgent
    {
        //protected TTSChannel tc;
        //protected string textToSpeak ="";
        //System.Threading.Thread newThread;
        //SpeechData currentSD;


        public VirtualHuman(AgentPlateform ap, string name, AgentClass agentclass)
            : base(ap, name, agentclass)
        {
            System.Console.WriteLine("VIRTUAL HUMAN");
        }

        public bool playAnimation(string animationName, int sens, double animationSpeed, bool cycle)// int sens = 1, double animationSpeed = 1.0, bool cycle = false
        {
            //return Body.playAnimation(animationName,sens,animationSpeed, cycle);
            return false;
        }

        public bool stopAnimation(string animationName)
        {
            //return Body.stopAnimation(animationName);
            return false;
        }

        public bool stopAllAnimations()
        {
            //return Body.stopAllAnimations();
            return false;
        }

        public bool playSound(string soundName, double pitch, bool cycle) // double pitch = 1.0, bool cycle = false
        {
            //return Body.playSound(soundName,pitch, cycle);
            return false;
        }

        bool stopSound(string soundName)
        {
            //return Body.stopSound(soundName);
            return false;
        }


        public double prepareSpeak(string text)
        {
            return Body.prepareSpeak(text);
        }
        public bool speak(string text)
        {
            return Body.speak(text);
           
        }

    }
}

